Core/Inc/cerb_utils.h

```diff
@@ -0,0 +1,27 @@
+/**
+ * @file cerb_utils.h
+ * @author Scott Abramson
+ * @brief Utility functions for Cerberus.
+ * @version 0.1
+ * @date 2024-08-04
+ * 
+ * @copyright Copyright (c) 2024
+ * 
+ */
+
+#ifndef CERB_UTILS_H
+#define CERB_UTILS_H
+
+#include "cmsis_os.h"
+#include "stdbool.h"
+
+/**
+ * @brief Function to debounce an input with an RTOS timer. Debounce is started and maintained by a high signal, and it is interrupted by a low signal.
+ * 
+ * @param input Input to debounce.
+ * @param timer RTOS timer with callback to be called at end of debounce period.
+ * @param period The period of time, in milliseconds, to debounce the input for.
+ */
+void debounce(bool input, osTimerAttr_t *timer, uint32_t period);
```

Comment:
could this be in embedded base eventually?

Reply:
Are we ok with putting an RTOS dependent function in embedded base? If the answer is yes, then yes this should be in embedded base.

Reply:
We can just make a new file called "RTOS utils" and put it there, or we can make it work with our own nertimer. I prefer the RTOS timers.

Reply:
Ya lets do that.